### ISSUE TITLE
Don't use ternary as an alternative for if else

### ISSUE THREAD

**Issue by Igor-Palaguta** [realm/SwiftLint#2358]
### Rule Request

1. Why should this rule be added? Share links to existing discussion about what
   the community thinks about this.

Ternary operator should be used just on right side of assignment or in return statement. And should not be used as an alternative for `if/else`. As it is very unusual and harder to read than usual `if/else`
https://softwareengineering.stackexchange.com/a/294560

2. Provide several examples of what _would_ and _wouldn't_ trigger violations.

```
// NOT TRIGGERED
let result = success ? "Success" : "Fail"

if success {
    askQuestion()
} else {
    exit()
}

var price: Double {
    return hasDiscount ? initialPrice - discount : initialPrice
}

// TRIGGERED
success ? askQuestion() : exit()

```

3. Should the rule be configurable, if so what parameters should be configurable?

No parameters expected

4. Should the rule be opt-in or enabled by default? Why?

It can be enabled by default

**Comment by drrost** [comment]
```
// TRIGGERED
success ? askQuestion() : exit()
```

I'd suggest not allow to call functions in the 2nd and 3rd part that return `void`.

**Comment by drrost** [comment]
> 4. Should the rule be opt-in or enabled by default? Why?
> 
> It can be enabled by default

And I'd make this rule disabled by default 'cause it's quite exotic behavior for C/Objective-C/Swift and many people may want to use it as they used to.
